### PR TITLE
Migrate Python doc to MVVM

### PR DESCRIPTION
**Pull request by eoogbe**

### CODE REVIEW

**Comment by eoogbe** [comment]
@lukesneeringer @shinfan PTAL

File: src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceDocTransformer.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import com.google.api.tools.framework.model.MessageType;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.ProtoFile;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import java.util.List;
+import java.util.Set;
+
+public class PythonGapicSurfaceDocTransformer implements ModelToViewTransformer {
```
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceDocTransformer.java]: nit: comments for the public class

File: src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceDocTransformer.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  private ViewModel generateDoc(
+      ProtoFile file, GapicProductConfig productConfig, ModelTypeTable typeTable) {
+    SurfaceNamer namer = new PythonSurfaceNamer(productConfig.getPackageName());
+
+    GrpcDocView.Builder doc = GrpcDocView.newBuilder();
+    doc.templateFileName(DOC_TEMPLATE_FILENAME);
+
+    String filename = file.getSimpleName();
+    doc.outputPath(filename.substring(0, filename.length() - ".proto".length()) + "_pb2.py");
```
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceDocTransformer.java]: FilenameUtils.removeExtension() or str.substring(0, str.lastIndexOf('.'))

File: src/main/java/com/google/api/codegen/transformer/py/PythonSurfaceNamer.java [hunk 3]
```diff
@@ -236,6 +243,12 @@ public String getFieldGetFunctionName(TypeRef type, Name identifier) {
   }
 
   @Override
+  public String getProtoFileName(ProtoFile file) {
+    String protoFilename = file.getSimpleName();
+    return protoFilename.substring(0, protoFilename.length() - "proto".length()) + "py";
```
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/py/PythonSurfaceNamer.java]: ditto

File: src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceDocTransformer.java (hunk 1 above)
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceDocTransformer.java]: done

File: src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceDocTransformer.java (hunk 2 above)
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/transformer/py/PythonGapicSurfaceDocTransformer.java]: done

File: src/main/java/com/google/api/codegen/transformer/py/PythonSurfaceNamer.java (hunk 3 above)
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/transformer/py/PythonSurfaceNamer.java]: done

**Pull request merged**